ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/HomeScreenMapContainerViewModel.kt

```diff
@@ -127,15 +129,16 @@ internal constructor(
       }
   }
 
-  private fun toLocationOfInterestFeatures(
+  private suspend fun toLocationOfInterestFeatures(
     locationsOfInterest: Set<LocationOfInterest>
   ): Set<Feature> = // TODO: Add support for polylines similar to mapPins.
   locationsOfInterest
       .map {
+        val submissionCount = submissionRepository.getSubmissions(it).size
```

Comment:
To do this for every LOI on the map will be expensive. Instead, Scott and I previously discussed caching the count in the LOI. I don't see it in the `LocationOfInterest` object, so I think that work still needs to be done. 

Would it be easier to set the value in a Cloud Function that gets triggered on write, or should the web and Android clients fetch the count and set the value on sync?

Reply:
I think setting the value in a Cloud Function triggered on write would probably be easier for the web and android clients. In both of those cases we're storing the submission count in multiple places rather than having a single source of truth which might get tricky making sure those are still in sync, but probably worthwhile for the loading optimization.

Reply:
Sgtm. Filed https://github.com/google/ground-platform/issues/1123 and assigned to myself. Would you be able to manually populate a new field on LOIs (`submissionCount`?) to test against that for now?